libraries/AP_Airspeed/AP_Airspeed.cpp

```diff
@@ -588,8 +588,12 @@ void AP_Airspeed::update_calibration(uint8_t i, float raw_pressure)
         if (state[i].cal.count == 0) {
             GCS_SEND_TEXT(MAV_SEVERITY_WARNING, "Airspeed %u unhealthy", i + 1);
         } else {
+            float calibrated_offset = state[i].cal.sum / state[i].cal.count;
             GCS_SEND_TEXT(MAV_SEVERITY_INFO, "Airspeed %u calibrated", i + 1);
-            param[i].offset.set_and_save(state[i].cal.sum / state[i].cal.count);
+            if (abs(calibrated_offset-param[i].offset) > 25) {
+                GCS_SEND_TEXT(MAV_SEVERITY_INFO, "Airspeed offset change too large,cover and reboot");
```

Comment:
This error messages suggest the value would not save, but it still does, so a second attempt would calibrate without warning. A message suggesting to reboot should probably also be sent at a higher severity level.

Maybe just: "Warning large change in airspeed offset" sent with warning severity.

Reply:
done...more or less